Source/WebCore/page/RemoteDOMWindow.cpp

```
@@ -113,13 +120,37 @@ WindowProxy* RemoteDOMWindow::parent() const
     return &m_frame->windowProxy();
 }
 
-void RemoteDOMWindow::postMessage(JSC::JSGlobalObject&, DOMWindow& incumbentWindow, JSC::JSValue message, const String& targetOrigin, Vector<JSC::Strong<JSC::JSObject>>&&)
+void RemoteDOMWindow::postMessage(JSC::JSGlobalObject& lexicalGlobalObject, DOMWindow& incumbentWindow, JSC::JSValue message, const String& targetOrigin, Vector<JSC::Strong<JSC::JSObject>>&& transfer)
 {
-    UNUSED_PARAM(incumbentWindow);
-    UNUSED_PARAM(message);
-    UNUSED_PARAM(targetOrigin);
-
-    // FIXME: Implemented this.
+    RefPtr sourceDocument = incumbentWindow.document();
+
+    // Compute the target origin.  We need to do this synchronously in order
+    // to generate the SyntaxError exception correctly.
+    std::optional<SecurityOriginData> target;
+    if (targetOrigin == "/"_s) {
+        if (!sourceDocument)
+            return;
+        target = sourceDocument->securityOrigin().data();
+    } else if (targetOrigin != "*"_s) {
+        target = SecurityOrigin::createFromString(targetOrigin)->data();
+        // It doesn't make sense target a postMessage at an opaque origin
+        // because there's no way to represent an opaque origin in a string.
+        if (target->isOpaque())
+            return; //Exception { SyntaxError };
```

If this should return an exception, why doesn't it?